game.py

```diff
@@ -14,13 +14,37 @@
 
 
 def snowman(snowman_word):
-    """Complete the snowman function
-    replace "pass" below with your own code
-    It should print 'Congratulations, you win!'
-    If the player wins and, 
-    'Sorry, you lose! The word was {snowman_word}' if the player loses
-    """
-    pass
+    # The word is provided as a parameter (tests pass it in)
+    snowman_word = snowman_word.lower()
+
+    # Build dict: each unique letter -> False (not guessed yet)
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+
+    # Track wrong guesses as a list
+    wrong_guesses_list = []
+
+    # Loop until too many wrong guesses OR the word is fully guessed
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES and not is_word_guessed(snowman_word, correct_letter_guess_statuses):
+        # Ask user for a letter (helper prevents repeats and validates)
+        user_input = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
+
+        if user_input in snowman_word:
```

Comment:
Note that the `if user_input in snowman_word` is (essentially) shorthand ([in and not in](https://docs.python.org/3/reference/expressions.html#in)) for: 
```py
for letter in snowman_word: 
    if user_guess == letter: 
        return True
    return False
```

You can use this syntax to perform the above logic over a multiple kinds of iterables/sequences in Python. Later in the curriculum we will learn that they don't all have the same performance (Big O).